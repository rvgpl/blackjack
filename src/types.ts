export interface Card {
  value: string;
  suit: string;
}

export enum CardType {
  Player,
  Dealer,
  Hidden,
}

export interface PickedCard extends Card {
  hidden: boolean;
}

export enum StatusMessage {
  PlayerWins = "You Win, Dealer Lose",
  DealerWins = "Dealer Wins, You Lose",
  Tie = "The game resulted in a Tie!",
  GameOver = "Game Over.",
}

export enum GameState {
  Init,
  PlayerHand,
  DealerHand,
  GameOver,
}
