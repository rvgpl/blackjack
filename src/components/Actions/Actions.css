.actions-container {
  display: grid;
  grid-gap: 10px;
  grid-template-columns: 1fr 1fr;
  position: absolute;
  width: calc(100% -16px);
  left: 8px;
  right: 8px;
  bottom: 16px;
}

.actions-button {
  background: var(--black);
  color: var(--white);
  padding: 8px 16px;
  display: flex;
  align-items: center;
  justify-content: center;
  border: none;
  text-transform: uppercase;
  font-size: var(--fs-m);
  letter-spacing: 1px;
  width: 100%;
  cursor: pointer;
  opacity: 1;
  transition: opacity 0.3 ease;
  outline: none;
}

.actions-button:hover {
  opacity: 0.8;
}

.actions-button:disabled {
  opacity: 0.3;
  cursor: not-allowed;
  pointer-events: none;
}

.action-button-full-width {
  grid-column: span 2;
}
