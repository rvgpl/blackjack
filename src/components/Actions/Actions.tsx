import React from "react";
import "./Actions.css";
import { GameState } from "../../types";

interface ActionsProps {
  gameState: GameState;
  hit: Function;
  stand: Function;
  reset: Function;
}

const Actions = ({ gameState, hit, stand, reset }: ActionsProps) => {
  return (
    <div className="actions-container">
      <button
        className="actions-button"
        disabled={
          (gameState === GameState.GameOver ||
            gameState === GameState.DealerHand) &&
          true
        }
        onClick={() => hit()}
      >
        Hit
      </button>
      <button
        className="actions-button"
        disabled={
          (gameState === GameState.GameOver ||
            gameState === GameState.DealerHand) &&
          true
        }
        onClick={() => stand()}
      >
        Stand
      </button>

      <button
        className="actions-button action-button-full-width"
        onClick={() => reset()}
      >
        Reset
      </button>
    </div>
  );
};

export default Actions;
