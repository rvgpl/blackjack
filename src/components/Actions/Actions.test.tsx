import React from "react";
import Actions from "./Actions";
import { shallow } from "enzyme";
import { GameState } from "../../types";

const hit = jest.fn();
const stand = jest.fn();
const reset = jest.fn();

describe("Actions", () => {
  it("should render all three buttons", () => {
    const wrapper = shallow(
      <Actions
        gameState={GameState.Init}
        hit={hit}
        reset={reset}
        stand={stand}
      />
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("button").length).toBe(3);
  });
});
