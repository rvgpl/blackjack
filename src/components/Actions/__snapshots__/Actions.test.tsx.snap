// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actions should render all three buttons 1`] = `ShallowWrapper {}`;
